packages/block-library/src/video/video-player.android.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Dashicon } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import { View, Linking, Alert } from 'react-native';
+
+/**
+ * Internal dependencies
+ */
+import { default as VideoPlayButton } from './video-play-button';
+import styles from './video-player.scss';
+
+class Video extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onPressPlay = this.onPressPlay.bind( this );
+	}
+
+	onPressPlay() {
+		const { source } = this.props;
+		if ( source && source.uri ) {
+			this.openURL( source.uri );
+		}
+	}
+
+	// Tries opening the URL outside of the app
+	openURL( url ) {
+		Linking.canOpenURL( url ).then( ( supported ) => {
+			if ( ! supported ) {
+				Alert.alert( __( 'Problem opening the video' ), __( 'No application can handle this request. Please install a Web browser.' ) );
+				window.console.warn( 'No application found that can open the video with URL: ' + url );
+			} else {
+				return Linking.openURL( url );
+			}
+		} ).catch( ( err ) => {
+			Alert.alert( __( 'Problem opening the video' ), __( 'An unknown error occurred. Please try again.' ) );
+			window.console.error( 'An error occurred while opening the video URL: ' + url, err );
+		} );
+	}
+
+	render() {
+		const { isSelected, style } = this.props;
+
+		return (
+			<View style={ styles.videoContainer }>
```

Comment:
On Gutenberg Web, there is an option to set a poster for the video. Probably we can use that prop `this.props.attributes.poster` and pass it here to show it as `<ImageBackground>` so that users can see the poster instead of black view?

Reply:
Interesting, that seems to be part of the extra block settings functionality though, to manually set a poster image. I can be wrong but, it doesn't seem useable for our case where we want to have a preview picture automatically shown, based on a frame from the video.

Reply:
Poster setting opened for public usage pretty recently, so I expect only a tiny amount of videos would come with a poster if any :(